### ISSUE TITLE
Train on my own datasets but error occurred: *** Check failure stack trace: ***

### ISSUE THREAD

**Issue by Cogito2012** [rbgirshick/fast-rcnn#95]
I had followed this help https://github.com/zeyuanxy/fast-rcnn/tree/master/help/train, but when I run the file train_net.py with argumemt:
     --solver models/CaffeNet/solver.prototxt --weights data/imagenet_models/CaffeNet.v2.caffemodel --imdb pku_bigmm_train
*** Check failure stack trace: *** occurred in train.py at line:
self.solver = caffe.SGDSolver(solver_prototxt)

/////////////
output in the console:
/////////////
Called with args:
Namespace(cfg_file=None, gpu_id=None, imdb_name='pku_bigmm_train', max_iters=40000, pretrained_model='data/imagenet_models/CaffeNet.v2.caffemodel', randomize=False, set_cfgs=None, solver='models/CaffeNet/solver.prototxt')
Using config:
{'DEDUP_BOXES': 0.0625,
 'EPS': 1e-14,
 'EXP_DIR': 'pku_bigmm',
 'PIXEL_MEANS': array([[[ 102.9801,  115.9465,  122.7717]]]),
 'RNG_SEED': 3,
 'ROOT_DIR': 'E:\\IIP-VS\\ObjectDetection\\Fast-RCNN\\fast-rcnn-bwt',
 'TEST': {'BBOX_REG': True,
          'MAX_SIZE': 1000,
          'NMS': 0.3,
          'SCALES': [600],
          'SVM': False},
 'TRAIN': {'BATCH_SIZE': 128,
           'BBOX_REG': True,
           'BBOX_THRESH': 0.5,
           'BG_THRESH_HI': 0.5,
           'BG_THRESH_LO': 0.1,
           'FG_FRACTION': 0.25,
           'FG_THRESH': 0.5,
           'IMS_PER_BATCH': 2,
           'MAX_SIZE': 1000,
           'SCALES': [600],
           'SNAPSHOT_INFIX': '',
           'SNAPSHOT_ITERS': 10000,
           'USE_FLIPPED': True,
           'USE_PREFETCH': False}}
Loaded dataset `pku_bigmm_train` for training
Appending horizontally-flipped training examples...
train ss roidb loaded from E:\IIP-VS\ObjectDetection\Fast-RCNN\fast-rcnn-bwt\data\cache\train_selective_search_roidb.pkl
done
Preparing training data...
done
Output will be saved to `E:\IIP-VS\ObjectDetection\Fast-RCNN\fast-rcnn-bwt\output\pku_bigmm\train`
Computing bounding-box regression targets...
done
*** Check failure stack trace: ***

**Comment by nw362** [comment]
Are there images without any annotations/GT boxes?

**Comment by Cogito2012** [comment]
@nw362 No such images, all my training images had at lest one annotations/GT box

**Comment by nw362** [comment]
A different issue then. It's a shame it doesn't give a file name and line number.

**Comment by Cogito2012** [comment]
@The error ocurred at the line 36 in the file ./lib/fast_rcnn/train.py
line 36: self.solver = caffe.SGDSolver(solver_prototxt)

**Comment by nw362** [comment]
I don't think it knows about the dataset until line 46 but I may be wrong. Does it work if you change dataset back to voc? 
line 46: self.solver.net.layers[0].set_roidb(roidb)

But just in case it's the same issue I had, apply 75b3fdd083f0490244441933b96581dcea719873 which gives a nicer error message.

**Comment by Cogito2012** [comment]
@nw362 The same error happened when I run the original version with voc2007 datasets. I doubt the problem is that I'm not using rbg's caffe. My caffe comes from @happyear. (https://github.com/happynear/caffe-windows), Mybe his caffe is not support for fast-rcnn training...or my caffe compiling is not right.

**Comment by nw362** [comment]
Probably, especially if it's missing a layer that the fast-rcnn-caffe has!
